openest/generate/functions.py

```diff
@@ -1148,3 +1148,61 @@ def describe():
                     arguments=[arguments.calculation, arguments.label.rename('name').optional(),
                                arguments.label.rename('title').optional(), arguments.description.optional()],
                     description="Change the name and/or description of a result.")
+
+class SequentialProcess(calculation.RecursiveCalculation):
+    def __init__(self, step1, step2, reportstep1=False):
+        if reportstep1:
+            super().__init__([step2, step1], step2.unitses + step1.unitses)
+        else:
+            super().__init__([step2, step1], step2.unitses)
+        self.step1 = step1
+        self.step2 = step2
+        self.reportstep1 = reportstep1
+
+    def format(self, lang, *args, **kwargs):
+        beforeauxlen = len(formatting.format_labels)
+        auxres = self.step1.format(lang, *args, **kwargs)
```

Comment:
`format` creates labels that can then be used by other `format` calls. This can be cleaned up and made more clear-- like so many changes, I had to add it in a rush way-back-when.